tekdoc/main.go

```diff
@@ -19,79 +19,92 @@ package main
 import (
 	"flag"
 	"fmt"
-	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
-	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned/scheme"
+	"io"
 	"io/ioutil"
 	"log"
 	"os"
 	"text/template"
-)
 
-var (
-	filename = flag.String("f", "", "Name of the file to parse")
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"github.com/tektoncd/pipeline/pkg/client/clientset/versioned/scheme"
 )
 
-func main() {
+// Take a file path, parse it as YAML -> v1beta1.Task
 
-	// TODO: take a file path, parse it as YAML -> v1alpha1.Task
-	flag.Parse()
-	dat, err := ioutil.ReadFile(*filename)
+func readTask(name string) (v1beta1.Task, error) {
+
+	var task v1beta1.Task
+	dat, err := ioutil.ReadFile(name)
 	if err != nil {
-		log.Fatalf("error reading file: %v", err)
+		return task, fmt.Errorf("error reading file: %w", err)
 	}
-
-	var task v1alpha1.Task
-
 	if _, _, err := scheme.Codecs.UniversalDeserializer().Decode(dat, nil, &task); err != nil {
-		log.Fatal(err)
+		return task, fmt.Errorf("error decoding task: %w", err)
 	}
 
-	// TODO: use Go templates to print v1alpha1.Task as Markdown
-	if task.Spec.Inputs != nil {
+	return task, nil
+}
+
+// Use Go templates to print v1beta1.Task as Markdown
+
+func printTask(w io.Writer, task v1beta1.Task) error {
+	if task.Spec.Params != nil {
 
 		tmpl := template.Must(template.New("test").Parse(`# {{.Name}}
 ## Install the Task
-kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/{{.Name}}/{{.Name}}.yaml
-### Input:-
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/master/task/{{.Name}}/0.1/{{.Name}}.yaml
```

Comment:
nit: We shouldn't assume the version number is always 0.1. Fine for now, but we should look into deriving this from the yaml in a follow up PR. :D